app/models/episode.rb

```diff
@@ -19,11 +28,26 @@ def update(args)
 
   class << self
     def prepare_params(args)
-      raise(ActiveRecord::RecordInvalid.new, 'Invalid video') unless args[:video].match VALID_YOUTUBE_LINK
+      if args[:video] && args[:date]
+        raise(ActiveRecord::RecordInvalid.new, 'Invalid video') unless args[:video].match VALID_YOUTUBE_LINK
 
-      args[:video] = args[:video].match(YOUTUBE_VIDEO_IDENTIFIER)[5]
-      args[:date] = Time.parse(args[:date]).utc
+        args[:video] = args[:video].match(YOUTUBE_VIDEO_IDENTIFIER)[5]
+        args[:date] = Time.parse(args[:date]).utc
+      end
       args
     end
   end
+
+  private
+
+  def check_youtube_status(video)
+    video.live_broadcast_content == 'none' ? 'over' : video.live_broadcast_content
+  end
+
+  def video_exists?(video)
+    @video = Yt::Video.new id: video
+    @video.title
+  rescue Yt::Errors::NoItems
```

Comment:
Again, why we need rescue in checker? Read about validations again pls

Reply:
I have read and tried it, but tests are still failing (without rescue) when I using any callbacks. Kirill said that he is recalling he request as unnecessary.